app/src/main/java/com/github/orkest/ui/sharing/SharingActivity.kt

```diff
@@ -69,6 +72,29 @@ class SharingComposeActivity : ComponentActivity() {
 
         super.onCreate(savedInstanceState)
 
+        // ----------------------------------------------------------------------------------------
+        // Check if device connected to the internet
+        if(!FireStoreDatabaseAPI.isOnline(this)){
+            Log.d("Debug", "No internet connection")
+            // alert the user that the device is not connected to the internet
+            val alertDialogBuilder = AlertDialog.Builder(this)
+            alertDialogBuilder.setTitle("No internet connection")
+            alertDialogBuilder.setMessage("Please connect to the internet to share songs with your friends")
+            alertDialogBuilder.setPositiveButton("OK") { _, _ ->
+                finish()
+            }
+            alertDialogBuilder.show()
+            // wait for the user to press OK
+            try {
+                Looper.loop()
+            } catch (_: RuntimeException) { }
```

Comment:
I would maybe suggest adding a Log when catching the exception to inform us that something happened.